src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  a?: number;
+  b?: number;
+  c?: number;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - You don't need to manually assign the constructor parameters to class properties when using TypeScript's parameter properties. You can remove the assignments and just use the access modifiers in the constructor parameters.